webapp-django/crashstats/crashstats/views.py

```diff
@@ -788,13 +789,16 @@ def crashes_per_day(request, default_context=None):
         supersearch_params['platform'].append('Mac')
         supersearch_params['platform'].remove('Mac OS X')
 
-    graph_data, results, adi_by_version = _get_crashes_per_day_with_adu(
-        supersearch_params,
-        start_date,
-        end_date,
-        platforms,
-        _date_range_type
-    )
+    try:
+        graph_data, results, adi_by_version = _get_crashes_per_day_with_adu(
+            supersearch_params,
+            start_date,
+            end_date,
+            platforms,
+            _date_range_type
+        )
+    except BadArgumentError as exception:
+        return http.HttpResponseBadRequest(unicode(exception))
```

Comment:
Why pass a unicode here? Methinks this should be a str. At a minimum, it'll get converted to a str so if you don't do the encoding yourself, then you leave it to Django to do in a way you don't control.